static/js/components/GroupSources.jsx

```diff
@@ -0,0 +1,250 @@
+import React, { useEffect, Suspense } from "react";
+import PropTypes from "prop-types";
+import { useSelector, useDispatch } from "react-redux";
+
+import TableCell from "@material-ui/core/TableCell";
+import TableRow from "@material-ui/core/TableRow";
+
+import Typography from "@material-ui/core/Typography";
+import IconButton from "@material-ui/core/IconButton";
+import Grid from "@material-ui/core/Grid";
+// import BuildIcon from "@material-ui/icons/Build";
+
+import Link from "@material-ui/core/Link";
+import PictureAsPdfIcon from "@material-ui/icons/PictureAsPdf";
+
+// import Menu from "@material-ui/core/Menu";
+// import MenuItem from "@material-ui/core/MenuItem";
+
+import MUIDataTable from "mui-datatables";
+import ThumbnailList from "./ThumbnailList";
+import styles from "./RunSummary.css";
+// import { observingRunTitle } from "./AssignmentForm";
+// import { ObservingRunStarList } from "./StarList";
+import * as SourcesAction from "../ducks/sources";
+// import * as Action from "../ducks/observingRun";
+import { ra_to_hours, dec_to_hours } from "../units";
+
+const VegaPlot = React.lazy(() => import("./VegaPlot"));
+
+const GroupSources = ({ route }) => {
+  const dispatch = useDispatch();
+  const sources = useSelector((state) => state.sources.latest);
+  const groups = useSelector((state) => state.groups.user);
+
+  // Load the group sources
+  useEffect(() => {
+    dispatch(SourcesAction.fetchSources({ group_id: route.id }));
```

Comment:
This could have side effects for other components that use `state.sources.latest` that don't re-fetch when they're rendered.

Reply:
I thought all components update as soon as the store is changed... maybe I got that wrong. But what is the solution, if I want to re-use the existing `fetchSources`?

Reply:
Yes, what I'm saying is that other components (specific `SourceList`) might expect `state.sources.latest` to initially be an un-filtered list (e.g. see `fetchSources` in `hydrate`) -- if so, that component will need to be updated to re-fetch upon render, otherwise it will display this filtered source list. Does that make sense?

Reply:
@guynir42 actually, I mis-wrote, the fetch doesn't occur in hydrate. But look at this check in `SourceList`: 
``` jsx
    if (!sources.latest) {
      dispatch(sourcesActions.fetchSources());
    }
```
This will obviously need to be updated so that it re-fetches upon every render. Another solution is just to add a new `groupSources` branch to the front-end state that this new component uses, so it won't mess up the `state.sources` branch that other components depend on.

Reply:
I can see there are a few ways to go about this. We should discuss it.

Reply:
I think the best approach is the latter noted above - either add a new branch to the front end state called `groupSources`, or another leaf to the `sources` branch with the same name.

Reply:
Probably the latter (add a new leaf `state.sources.groupSources`)

Reply:
Ok this is now added to sources.groupSources in the state tree.